### PR TITLE
feat: update price warning modal

### PR DESCRIPTION
**Pull request by viet-nv**

### CODE REVIEW

File: src/components/SwapForm/SwapModal/SwapDetails/index.tsx [hunk 1]
```diff
@@ -255,6 +258,23 @@ const SwapDetails: React.FC<Props> = ({
...
+          <RowFixed>
+            <TYPE.black fontSize={14} fontWeight={400} color={theme.subText}>
+              <Trans>Price Impact</Trans>
+            </TYPE.black>
+            <InfoHelper size={14} text={t`Estimated change in price due to the size of your transaction`} />
+          </RowFixed>
+          <TYPE.black
+            fontSize={14}
+            color={
+              priceImpact ? (priceImpact > 15 ? theme.red : priceImpact > 5 ? theme.warning : theme.text) : theme.text
```
Comment (nguyenhoaidanh) [inline src/components/SwapForm/SwapModal/SwapDetails/index.tsx:271]: use 
`const { isInvalid,isHigh,isVeryHigh} = checkPriceImpact(priceImpact)`

File: src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx [hunk 2]
```diff
@@ -53,6 +75,12 @@ const ConfirmSwapModalContent: React.FC<Props> = ({
 
   const shouldDisableConfirmButton = isBuildingRoute || !!errorWhileBuildRoute
 
+  const priceImpactFromBuild =
```
Comment (hungdoansy) [inline src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx:78]: Needs a TODO here. We'll implement a common function to calculate price impact later

File: src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx [hunk 3]
```diff
@@ -112,6 +142,16 @@ const ConfirmSwapModalContent: React.FC<Props> = ({
...
+  const theme = useTheme()
+  const warningStyle = priceImpactResult.isVeryHigh
+    ? { background: theme.red, color: theme.black }
+    : priceImpactResult.isHigh
+    ? { background: theme.warning, color: theme.black }
+    : undefined
+
+  const [confirmNewPrice, setConfirmNewPrice] = useState(false)
+
+  const disableSwap = (outputAmountChange < 0 && !confirmNewPrice) || shouldDisableConfirmButton
```
Comment (hungdoansy) [inline src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx:155]:
```suggestion
  const shouldDisableSwapButton = (outputAmountChange < 0 && !confirmNewPrice) || shouldDisableConfirmButton
```

File: src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx [hunk 4]
```diff
@@ -53,6 +75,13 @@ const ConfirmSwapModalContent: React.FC<Props> = ({
 
   const shouldDisableConfirmButton = isBuildingRoute || !!errorWhileBuildRoute
 
+  const priceImpactFromBuild =
+    15.1 ||
```
Comment (hungdoansy) [inline src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx:79]: Redundant

File: src/components/SwapForm/SwapModal/SwapDetails/index.tsx (hunk 1 above)
Comment (viet-nv) [reply src/components/SwapForm/SwapModal/SwapDetails/index.tsx:271]: updated

**Comment by nguyenhoaidanh** [comment]
lgtm, @hungdoansy  can approve

**Review by nguyenhoaidanh — DISMISSED** [review]

File: src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx (hunk 4 above)
Comment (viet-nv) [reply src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx]: just for testing, I updated right after I push it. LOL

File: src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx (hunk 2 above)
Comment (viet-nv) [reply src/components/SwapForm/SwapModal/ConfirmSwapModalContent.tsx:77]: no need for now. more things need to do in next week

**Review by hungdoansy — DISMISSED** [review]

**Review by nguyenhuudungz — APPROVED** [review]

**Pull request merged**